go-controller/pkg/ovn/egressip.go

```diff
@@ -1221,14 +1268,30 @@ func (oc *Controller) syncStaleSNATRules(egressIPToPodIPCache map[string]sets.St
 // atomic items with the same general information repeated across most (egressIP
 // name, logical IP defined for that name), hence use a cache to avoid round
 // trips to the API server per item.
-func (oc *Controller) generatePodIPCacheForEgressIP(eIPs []interface{}) (map[string]sets.String, error) {
+func (oc *Controller) generatePodIPCacheForEgressIP(eIPs []interface{}) (map[string]sets.String, map[string]sets.String, map[string]sets.String, error) {
 	egressIPToPodIPCache := make(map[string]sets.String)
+	egressIPToPodIPCacheNextHops := make(map[string]sets.String)
+	egressIPToPodIPCacheNodeIP := make(map[string]sets.String)
 	for _, eIP := range eIPs {
 		egressIP, ok := eIP.(*egressipv1.EgressIP)
 		if !ok {
 			continue
 		}
 		egressIPToPodIPCache[egressIP.Name] = sets.NewString()
+		egressIPToPodIPCacheNextHops[egressIP.Name] = sets.NewString()
+		egressIPToPodIPCacheNodeIP[egressIP.Name] = sets.NewString()
+
+		for _, status := range egressIP.Status.Items {
+			isEgressIPv6 := utilnet.IsIPv6String(status.EgressIP)
+			gatewayRouterIP, err := oc.eIPC.getGatewayRouterJoinIP(status.Node, isEgressIPv6)
+			if err != nil {
+				klog.Errorf("Unable to retrieve gateway IP for node: %s, protocol is IPv6: %v, err: %v", status.Node, isEgressIPv6, err)
+				continue
+			}
+			egressIPToPodIPCacheNextHops[egressIP.Name].Insert(gatewayRouterIP.String())
```

Comment:
should the name of this map just be called "egressIPToGRIPCache" ?

Reply:
how about: `gatewayRouterIPs` ? 
```
egressIPCache[egressIP.Name].gatewayRouterIPs.Insert(gatewayRouterIP.String())
```